packages/worker/src/sdk/users/users.ts

```diff
@@ -265,6 +280,11 @@ export const save = async (
     await eventHelpers.handleSaveEvents(builtUser, dbUser)
     await addTenant(tenantId, _id, email)
     await cache.user.invalidateUser(response.id)
+
+    for (const appId of appsToRemove) {
+      await apps.removeUserFromApp(_id, appId)
+    }
```

Comment:
The looping network call might have some performance problems if this request is ever used with more than one app. I think we could get the server to perform this logic, as a result of the existing `syncUserInApps` call. It would be nice to contain all the user -> app user sync in one place like this. We already have some logic in there for when the user is actually being deleted. 

I assume it is not working in this case as the roles being removed from the user are already saved to the db at the time of the request being sent and so the server never knows to delete the relationship. We could update the existing sync to receive a body: 
```
{
   "previousUser": {} // supply the `dbUser` if it exists
}
```

Then in the sync code here:
https://github.com/Budibase/budibase/blob/develop/packages/server/src/api/controllers/user.ts#L14
we can perform a diff to get the list of removed apps and action on them, like is being done above.

Reply:
I like the idea :)